web/pgadmin/misc/bgprocess/processes.py

```diff
@@ -608,6 +630,10 @@ def status(self, out=0, err=0):
                 'process_state': self.process_state
             }
 
+        # Set error message based on exit code if stderr is empty.
+        if err_completed and len(stderr) == 0:
```

Comment:
There is no harm in putting the error message even if there is something in stderr. It may be possible that existing stderr doesn't have the error details. I should simply check `err_completed and self.ecode != 0`

Reply:
I did that first, but somehow it ended up stuck in an infinite loop.